validator/client/validator_propose.go

```diff
@@ -0,0 +1,77 @@
+package client
+
+// Validator client proposer functions.
+
+import (
+	"context"
+
+	ptypes "github.com/gogo/protobuf/types"
+	"github.com/opentracing/opentracing-go"
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/ssz"
+)
+
+// ProposeBlock
+//
+// WIP - not done.
+func (v *validator) ProposeBlock(ctx context.Context, slot uint64) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "validator.ProposeBlock")
+	defer span.Finish()
+
+	// 1. Fetch data from Beacon Chain node.
+	// Get current head beacon block.
+	headBlock, err := v.beaconClient.CanonicalHead(ctx, &ptypes.Empty{})
+	if err != nil {
+		log.Errorf("Failed to fetch CanonicalHead: %v", err)
+		return
+	}
+	_ = headBlock // TODO(1461): Actually tree hash the block.
+	parentTreeHash, err := ssz.TreeHash(false)
+	if err != nil {
+		log.Errorf("Failed to hash parent block: %v", err)
+		return
+	}
+
+	// Get pending deposit data.
```

Comment:
A bit more informative comment, perhaps: "fetch validator deposits into the ETH1.0 which have not yet been included in the beacon chain"